### PR TITLE
yarsi: init at 0.1.0

### PR DESCRIPTION
**Pull request by transgirllucy**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by QJoly** [comment]
(Be careful, I'm a noob)

You need to squash your commits

File: pkgs/tools/misc/yarsi/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  pname = "yarsi";
+  version = "0.1.0";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-jyiKB2VdCRUzLjD8XaFQOIkgI6gE8uM0ztx4vqd6SrM=";
+  };
+
+  cargoSha256 = "sha256-yr43w53SrdFR2LhYwt6lh4mdNV98vc8XEAFn2JUsfn8=";
+
```
Comment (AndersonTorres) [inline pkgs/tools/misc/yarsi/default.nix:13]:
```suggestion
  cargoHash = "sha256-yr43w53SrdFR2LhYwt6lh4mdNV98vc8XEAFn2JUsfn8=";
```

File: pkgs/tools/misc/yarsi/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ lib, rustPlatform, fetchCrate }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "yarsi";
+  version = "0.1.0";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-jyiKB2VdCRUzLjD8XaFQOIkgI6gE8uM0ztx4vqd6SrM=";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/yarsi/default.nix:9]:
```suggestion
    hash = "sha256-jyiKB2VdCRUzLjD8XaFQOIkgI6gE8uM0ztx4vqd6SrM=";
```

File: pkgs/tools/misc/yarsi/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  };
+
+  cargoSha256 = "sha256-yr43w53SrdFR2LhYwt6lh4mdNV98vc8XEAFn2JUsfn8=";
+
+
+  meta = with lib; {
+    description = "Yet another rust system info fetch. ";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/yarsi/default.nix:16]:
```suggestion
    description = "Yet another rust system info fetch";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -40060,4 +40060,6 @@ with pkgs;
   zf = callPackage ../tools/misc/zf { };
 
   isolate = callPackage ../tools/security/isolate { };
+
+  yarsi = callPackage ../tools/misc/yarsi { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:40064]: Please sort this entry into a category and there alphabetically.

File: pkgs/tools/misc/yarsi/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+  src = fetchCrate {
+    inherit pname version;
+    hash = "sha256-jyiKB2VdCRUzLjD8XaFQOIkgI6gE8uM0ztx4vqd6SrM=";
+  };
+
+  cargoHash = "sha256-yr43w53SrdFR2LhYwt6lh4mdNV98vc8XEAFn2JUsfn8=";
+
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/yarsi/default.nix:15]:
```suggestion
  cargoHash = "sha256-yr43w53SrdFR2LhYwt6lh4mdNV98vc8XEAFn2JUsfn8=";

  meta = with lib; {
```

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:40064]: Merge this commit with the other.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/yarsi/default.nix (hunk 2 above)
Comment (AndersonTorres) [inline pkgs/tools/misc/yarsi/default.nix:9]:
```suggestion
    hash = "sha256-jyiKB2VdCRUzLjD8XaFQOIkgI6gE8uM0ztx4vqd6SrM=";
```

File: pkgs/tools/misc/yarsi/default.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  cargoSha256 = "sha256-yr43w53SrdFR2LhYwt6lh4mdNV98vc8XEAFn2JUsfn8=";
+
+
+  meta = with lib; {
+    description = "Yet another rust system info fetch. ";
+    homepage = "https://github.com/bytehunt/yarsi";
+    license = licenses.mit;
+    maintainers = with maintainers; [ mrtuxa ];
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/tools/misc/yarsi/default.nix:21]:
```suggestion
}

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Remember to cleanup whitespaces.

Run `git diff --check` to see them.

File: pkgs/tools/misc/yarsi/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/tools/misc/yarsi/default.nix:13]:
```suggestion

  buildInputs = with darwin.apple_sdk.frameworks; [ SystemConfiguration ];

```

File: pkgs/tools/misc/yarsi/default.nix [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{ lib, rustPlatform, fetchCrate }:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/yarsi/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchCrate, darwin }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**